pkg/circuit_breaker/store.go

```diff
@@ -0,0 +1,198 @@
+package circuit_breaker
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/frain-dev/convoy/pkg/clock"
+	"github.com/go-redsync/redsync/v4"
+	"github.com/go-redsync/redsync/v4/redis/goredis/v9"
+	"github.com/redis/go-redis/v9"
+	"strings"
+	"sync"
+	"time"
+)
+
+type CircuitBreakerStore interface {
+	Lock(ctx context.Context, lockKey string) (*redsync.Mutex, error)
```

Comment:
@jirevwe Should we couple the store implementation with the distributed lock?

Reply:
I don't think it's a bad idea for now

Reply:
Fair. It was only a nitpick. :)